Toolset/libraries/revmobilelibrary.livecodescript

```diff
@@ -0,0 +1,621 @@
+         put "Update failed on the SQL command:" && pError & cr & "Revdb error:" && tResult into tErrorMessage
+         break
+      case "NoKey"
+         put "No primary key has been entered for the database:" && short name of pError & cr & "Revdb error:" && tResult into tErrorMessage
+         break
+      case "NoTable"
+         put "No table has been selected for updating the database: " && short name of pError  & cr & "Revdb error:" && tResult into tErrorMessage
+         break
+      case "NoQuery"
+         put "No SQL command set in the database:" && short name of pError into tErrorMessage
+         break
+      case "ExecuteQuery"
+         put "An error resulted for the SQL command in the database:" && short name of pError & cr & "Revdb error:" && tResult into tErrorMessage
+         break 
+      case "EmptyView"
+         put "No records returned in the database:" && short name of pError into tErrorMessage
+         break
+      default
+         put "An error has occurred with the database:" && short name of pError & cr & "Revdb error:" && tResult into tErrorMessage
+   end switch
+   
+   answer error tErrorMessage
+end revDatabaseError
+
+--function revAnswerFile pType, pPrompt
+
+--function revAnswerFiles pType, pPrompt
+
+--command scrollUp
+
+--command scrollDown
+
+--command scrollLeft
+
+--command scrollRight
+
+--command revFontApplySystemStyle pObject, pSize, pStyles
+
+--function revFontGetSystemFont
+
+--function revFontGetSystemFontSize
+
+--function  revFontGetSystemFontStyles
+
+--private command _revFontApplyStylesToObject pObject, pFont, pSize, pStyle
+
+--private function _revFontGetPlatform
+
+--private function _revEnvironmentPlatform
+
+# Parameters
+#   pPath : Any valid path that can be ensured
+# Returns
+#   empty if successful, an error string otherwise
+# Description
+#   Makes sure that a path exists by creating any folders that are required.
+#   Will take a path including a filename, but will not create the file, only
+#   the folders that contain it.
+command revEnsurePath pPath
+   if pPath is empty then
+      exit revEnsurePath
+   end if
+   
+   if there is a folder pPath or there is a file pPath then
+      exit revEnsurePath
+   end if
+   
+   local tPath, tPart
+   set the itemDelimiter to slash
+   repeat for each item tPart in pPath
+      put tPart & "/" after tPath
+      if there is no folder tPath then
+         create folder tPath
+      end if
+   end repeat
+end revEnsurePath
+
+--function revLicenseType
+
+#--function macVersionLessThan pLeft, pRight
+
+# Parameters
+#   pObject : either a long id or a valid reference to an object.
+# Returns
+#   The "rugged" id of the object. This can be used the same as a long id,
+#   but is slightly more resistant to stuff like objects being grouped, renamed etc.
+#   The script editor uses rugged ids to store object ids when there is a chance that the object
+#   could be renamed or grouped etc. 
+function revRuggedId pObject
```

Comment:
Reading comments in the code, it appears as if this function is not really needed except for inside the IDE (there is even at least one case where a check for development environment is done and this is only called if so).  If that is indeed true, then this function could be stripped down for the mobile library and just return the long id of the object.

Reply:
@bwmilby the remote debugger needs this

Reply:
On thing is that this code is not currently available on mobile unless remote debugger pushes the common library somehow.

Reply:
yes it adds it in as a dependency along with the debugger library and the metadata library

Reply:
Looking at that code (debugger library), i can see the call, but is the whole function really needed on mobile?  There are places in code where if not in development environment then the long ID is used.  Reading the comments looks like the code is needed to account for objects in groups while in edit group mode.  My guess is that at some point it was added to the common library due to multiple dependencies (profile library?).  Could it be simplified in the common library to check for development and pass and if not just return long ID?
```
# simplified function for standalones where edit group mode can't cause a problem
function revRuggedId pObject
   if the environment is "development" then
      pass revRuggedID -- let revBackscript handle the call
   else
      return the long id of pObject
   end if
end revRuggedId
```

Reply:
It is used considerably more by the remote debugger.